src/persistence/parsers/index.ts

```diff
@@ -77,28 +77,33 @@ const pendingDeposit: z.ZodSchema<PendingDeposit> = z.object({
   transactionId: z.string().optional(),
 });
 
-const pendingWithdraw: z.ZodSchema<PendingWithdraw> = z.object({
-  accountIndex: z.string(),
-  amount: z.string(),
-  batchNum: z.number(),
-  hash: z.string(),
-  id: z.string(),
-  timestamp: z.string(),
-  token,
-});
-
 const merkleProof: z.ZodSchema<MerkleProof> = z.object({
   root: z.string(),
   siblings: z.string().array(),
 });
 
-const pendingDelayedWithdraw: z.ZodSchema<PendingDelayedWithdraw> = pendingWithdraw.and(
+const pendingWithdraw: z.ZodSchema<PendingWithdraw> = hermezApiResourceItem.and(
```

Comment:
I think we should compose the parsers as we compose the types:

```typescript
const exit: z.ZodSchema<Exit> = hermezApiResourceItem.and(
  z.object({
    accountIndex: z.string(),
    batchNum: z.number(),
    delayedWithdraw: z.number().nullable(),
    instantWithdraw: z.number().nullable(),
    token,
    merkleProof,
    balance: z.string(),
  })
);

const pendingWithdraw: z.ZodSchema<PendingWithdraw> = exit.and(
  z.object({
    hermezEthereumAddress: z.string(),
    hash: z.string(),
    id: z.string(),
    timestamp: z.string(),
  })
);
```

Notice also the syntax `z.number().nullable()` which I think is more readable that `z.nullable(z.number())`

Reply:
Take into account that `Exit` is a concept from `hermezjs` that we are not parsing yet. That's why I decided to keep it as just `pendingWithdraw` and `pendingDelayedWithdraw` (domain from the wallet as now).

But I agree that later when we add the parsing from the library the parsers should be as you are proposing!

PS: `nullable` syntax fixed!